data_generation/expand.py

```diff
@@ -27,49 +27,88 @@ def get_args():
     )
     # if file is moved in another directory level relative to the root (currently in root/data_generation), this needs to be changed
     root_directory = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-    parser.add_argument("-d", "--data_path", default=root_directory + "/Data/train")
-    parser.add_argument("-s", "--save_dir", default=root_directory + "/Data/expand")
+    parser.add_argument("-d", "--data_path", default=root_directory + "/Data")
+    parser.add_argument("--dataset", choices=['train', 'val', 'test'], default='train')
+    parser.add_argument("-s", "--save_dir", default=root_directory + "/Data")
+    parser.add_argument("-n", "--num_networks", type=int, default=1)
     args = parser.parse_args()
     print(args)
     return args
 
 
 def expand(args):
+    """
+    Expands the given power network dataset by creating new networks with variations in load and generator power.
+
+    Args:
+        args: An object containing the following attributes:
+            - data_path: The path to the directory containing the input dataset.
+            - dataset: The name of the dataset to be expanded.
+            - num_networks: The number of new networks to be generated for each input network.
+            - save_dir: The path to the directory where the expanded dataset will be saved.
+
+    Returns:
+        n: The total number of new networks generated.
+        time: The time taken to generate the new networks.
+    """
     start = time.perf_counter()
-    p = args.data_path
+    p = f"{args.data_path}/{args.dataset}"
+    print(f"Loading networks from {p}")
     graph_paths = sorted(os.listdir(f"{p}/x"))
     print(f"Num paths: {len(graph_paths)}")
     n = 0
     for g in tqdm.tqdm(graph_paths):
-        graph = pp.from_json(f"{p}/x/{g}")
-        for _ in range(int(len(graph.bus) / 4)):
-            new_graph = deepcopy(graph)
-            for i in graph.gen.index:
-                new_graph.gen.loc[i, 'p_mw'] = np.random.normal(graph.gen.loc[i, 'p_mw'], math.sqrt(abs(graph.gen.loc[i, 'p_mw'])))
-            for i in graph.load.index:
-                new_graph.load.loc[i, 'p_mw'] = np.random.normal(graph.load.loc[i, 'p_mw'], math.sqrt(abs(graph.load.loc[i, 'p_mw'])))
-                new_graph.load.loc[i, 'q_mvar'] = np.random.normal(graph.load.loc[i, 'q_mvar'], math.sqrt(abs(graph.load.loc[i, 'q_mvar'])))
+        for _ in range(args.num_networks):
+            graph = pp.from_json(f"{p}/x/{g}")
+            for _ in range(int(len(graph.bus) / 4)):
```

Comment:
I think the 2nd for loop `for _ in range(int(len(graph.bus) / 4)):` should be removed and `graph = pp.from_json(f"{p}/x/{g}")` would be better outside of the` for _ in range(args.num_networks)`. With the current two for loops we have here, we generate `num_networks * int(len(graph.bus) / 4)` varied graphs for each graph

Reply:
changes discussed have been added